File: reverseinteger.java
public class Solution {
    public int reverse(int x) {
        // Start typing your Java solution below
        // DO NOT write main() function
        int k = Math.abs(x);
       
        int r = 1;
        int ans = 0;
        while(k>=10){ // !!!!!!
            r = k%10;
            ans *=10;
            ans += r;
            k/=10;
        }
        ans *=10;
        ans +=k;
        if(x<0) ans = -ans;
        return ans;
    }
}

File: ReverseInteger.cpp
class Solution {
public:
    int reverse(int x1) {
        bool pos = (x1>=0);
        long long int x = x1;
        if(!pos) {
            x = -x;
        }
        long long int y = 0;
        while(x>=10) {
            y += x%10;
            y *= 10;
            x = x/10;
        }
        y += x;
        if (y > 2147483647LL) {
            return 0;
        }
        if(!pos) {
            y = -y;
        }
        return y;
    }
};
